src/firefox/index.js

```diff
@@ -50,27 +50,39 @@ export function defaultRemotePortFinder(
   {
     portToTry=REMOTE_PORT,
     connectToFirefox=defaultFirefoxConnector,
-  }: RemotePortFinderParams = {}
+  }: RemotePortFinderParams = {},
+
 ): Promise<number> {
   log.debug(`Checking if remote Firefox port ${portToTry} is available`);
-
-  return connectToFirefox(portToTry)
-    .then((client) => {
-      log.debug(`Remote Firefox port ${portToTry} is in use`);
-      client.disconnect();
-      // TODO: instead of throw an error, pick a new random port until
-      // one of them is available.
-      // https://github.com/mozilla/web-ext/issues/283
-      throw new WebExtError(
-        `Cannot listen on port ${portToTry} because it's in use`);
+  let retriesCount = 10;
+  function portConnectionRetry(){
+    return new Promise((resolve, reject) => {
+      connectToFirefox(portToTry)
+        .then((client) => {
+          log.debug(`Remote Firefox port ${portToTry} is in use.`);
+          client.disconnect();
+          portToTry++;
+          if (retriesCount > 0){
+            retriesCount--;
+            return portConnectionRetry();
+          }
+          else {
+            throw new WebExtError('Too many retries on port search');
+          }
+        })
+       .catch(onlyErrorsWithCode('ECONNREFUSED', () =>{
+         resolve(portToTry);
+       }))
+       .catch((err) => {
+         reject(err);
+       });
     })
-    .catch(onlyErrorsWithCode('ECONNREFUSED', () => {
-      // The connection was refused so this port is good to use.
-      return portToTry;
-    }));
+    .then((port) => {
+      return port;
+    });
+  }
+  return portConnectionRetry();
```

Comment:
instead of `portConnectionRetry()`, I think a more concise and descriptive name would be `tryToFindAnOpenPort()`